server/roomlogs.js

```diff
@@ -90,16 +93,18 @@ class Roomlog {
 		log = [];
 		for (let i = 0; i < this.log.length; ++i) {
 			const line = this.log[i];
-			if (line === '|split') {
-				const ownLine = this.log[i + channel + 1];
+			const split = /\|split\|p(\d)/g.exec(line);
+			if (split) {
+				const offset = (channel === OMNISCIENT_CHANNEL || Number(split[0]) === channel) ? 1 : 2;
```

Comment:
I would probably not draw the line there.

```suggestion
				const canSeePrivileged = (channel === Number(split[0]) || channel === OMNISCIENT_CHANNEL);
```